shap/plots/_violin.py

```diff
@@ -124,94 +131,12 @@ def violin(shap_values, features=None, feature_names=None, max_display=None, plo
     if use_log_scale:
         pl.xscale('symlog')
 
-    # plotting SHAP interaction values
-    if not multi_class and len(shap_values.shape) == 3:
-
-        if plot_type == "compact_dot":
-            new_shap_values = shap_values.reshape(shap_values.shape[0], -1)
-            new_features = np.tile(features, (1, 1, features.shape[1])).reshape(features.shape[0], -1)
-
-            new_feature_names = []
-            for c1 in feature_names:
-                for c2 in feature_names:
-                    if c1 == c2:
-                        new_feature_names.append(c1)
-                    else:
-                        new_feature_names.append(c1 + "* - " + c2)
-
-            return summary(
-                new_shap_values, new_features, new_feature_names,
-                max_display=max_display, plot_type="dot", color=color, axis_color=axis_color,
-                title=title, alpha=alpha, show=show, sort=sort,
-                color_bar=color_bar, plot_size=plot_size, class_names=class_names,
-                color_bar_label="*" + color_bar_label
-            )
-
-        if max_display is None:
-            max_display = 7
-        else:
-            max_display = min(len(feature_names), max_display)
-
-        sort_inds = np.argsort(-np.abs(shap_values.sum(1)).sum(0))
-
-        # get plotting limits
-        delta = 1.0 / (shap_values.shape[1] ** 2)
-        slow = np.nanpercentile(shap_values, delta)
-        shigh = np.nanpercentile(shap_values, 100 - delta)
-        v = max(abs(slow), abs(shigh))
-        slow = -v
-        shigh = v
-
-        pl.figure(figsize=(1.5 * max_display + 1, 0.8 * max_display + 1))
-        pl.subplot(1, max_display, 1)
-        proj_shap_values = shap_values[:, sort_inds[0], sort_inds]
-        proj_shap_values[:, 1:] *= 2  # because off diag effects are split in half
-        summary(
-            proj_shap_values, features[:, sort_inds] if features is not None else None,
-            feature_names=feature_names[sort_inds],
-            sort=False, show=False, color_bar=False,
-            plot_size=None,
-            max_display=max_display
-        )
-        pl.xlim((slow, shigh))
-        pl.xlabel("")
-        title_length_limit = 11
-        pl.title(shorten_text(feature_names[sort_inds[0]], title_length_limit))
-        for i in range(1, min(len(sort_inds), max_display)):
-            ind = sort_inds[i]
-            pl.subplot(1, max_display, i + 1)
-            proj_shap_values = shap_values[:, ind, sort_inds]
-            proj_shap_values *= 2
-            proj_shap_values[:, i] /= 2  # because only off diag effects are split in half
-            summary(
```

Comment:
The reference to the "legacy" `summary` function here is the offending line for Undefined Name error.

Note that no `plot_type` is specified in this function call, meaning that it was intended to be a "dot" plot. Hence, not relevant for violin plot.